transform/app/validations/solr.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from typing import List
+
+from app.services.solr.configsets import get_configset_names
+from app.services.solr.collections import get_collection_names
+
+logger = logging.getLogger(__name__)
+
+
+def validate_configset_exists(configset: str) -> None:
+    """
+    Validates the existence of a Solr configset.
+
+    Parameters:
+        configset (str): The name of the Solr configset to be validated.
+
+    Raises:
+        ValueError: If the specified configset does not exist.
+    """
+    existing_configsets = get_configset_names()
+    if configset not in existing_configsets:
+        error_msg = f"configset {configset} does not exist. Existing configsets: {existing_configsets}"
+        logger.error(error_msg)
+        raise ValueError(error_msg)
+
+
+def validate_collections_not_exist(collection_names: List[str]) -> None:
+    """
+    Validates that specified Solr collections do not already exist.
+
+    Parameters:
+        collection_names (List[str]): A list of collection names to be validated.
+
+    Raises:
+        ValueError: If any of the specified collections already exist.
+    """
+    existing_collections = get_collection_names()
+
+    not_available_collections = [
+        collection
+        for collection in collection_names
+        if collection in existing_collections
+    ]
+
+    if not_available_collections:
+        error_msg = f"Some collections in this iteration already exist: {not_available_collections}"
```

Comment:
Changed, I also changed this function because I will use the opposite of it in the next PR with a different endpoint. Final message does not contain "Creating collections failed" because I will use this code for other purposes.